PriorityQueue/PriorityQueue/PriorityQueue.cs

```diff
@@ -0,0 +1,84 @@
+﻿namespace PriorityQueue;
+
+/// <summary>
+/// Priority Queue realization on list with numeric priority and generic value
+/// </summary>
+public class PriorityQueue<T>
+{
+    /// <summary>
+    /// Pointer to the head of the queue
+    /// </summary>
+    private QueueElement? head;
+    private int count;
+
+    /// <summary>
+    /// Adds a value to the queue by priority
+    /// </summary>
+    /// <param name="value">Value to add</param>
+    /// <param name="priority">Priority value</param>
+    public void Enqueue(T value, int priority)
+    {
+        var currentElement = head;
+        var newElement = new QueueElement(value, priority);
+
+        if (head == null || head.Priority < newElement.Priority)
+        {
+            newElement.Next = head;
+            head = newElement;
+            count++;
+            return;
+        }
+
+        while (currentElement?.Next != null && currentElement.Next.Priority >= newElement.Priority)
+        {
+            currentElement = currentElement.Next;
+        }
+        
+        count++;
+        newElement.Next = currentElement?.Next;
+        currentElement!.Next = newElement;
+    }
+ 
+    /// <summary>
+    /// Returns the value of the queue head and moves the head to the next element
+    /// </summary>
+    public T Dequeue()
+    {
+        if (head == null)
+        {
+            throw new InvalidOperationException();
+        }
+
+        var value = head.Value;
+        head = head.Next;
+        count--;
+        return value;
+    }
+
+    
+    /// <summary>
+    /// IsEmpty method
+    /// </summary>
+    public bool IsEmpty()
+    {
+        return count <= 0;
+    }
```

Comment:
```suggestion
    public bool IsEmpty()
        => count <= 0;
```
Или лучше `public bool Empty => count <= 0;`. Если метод суть запрос состояния объекта и не требует серьёзных вычислений, его лучше делать свойством.